editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/EventsLog.tsx

```diff
@@ -0,0 +1,170 @@
+
+const ErrorSummary = styled(Box)(({ theme }) => ({
+  marginTop: theme.spacing(1),
+  padding: theme.spacing(3),
+  border: `5px solid ${theme.palette.error.main}`,
+}));
+
+const EventsLog: React.FC<GetSubmissionsResponse> = ({
+  submissions,
+  loading,
+  error,
+}) => {
+  if (loading)
+    return (
+      <DelayedLoadingIndicator
+        msDelayBeforeVisible={0}
+        text="Fetching events..."
+      />
+    );
+  if (error) return <ErrorFallback error={error} />;
+  if (submissions.length === 0)
+    return (
+      <ErrorSummary role="status" data-testid="error-summary">
+        <Typography variant="h4" gutterBottom>
+          {`No payments or submissions found for this service`}
+        </Typography>
+        <Typography variant="body2">
+          {`If you're looking for events before January 1, 2024, please contact a PlanX developer.`}
+        </Typography>
+      </ErrorSummary>
+    );
+
+  return (
+    <TableContainer>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell sx={{ width: 300 }}>
+              <strong>Event</strong>
+            </TableCell>
+            <TableCell sx={{ width: 200 }}>
+              <strong>Date</strong>
+            </TableCell>
+            <TableCell>
+              <strong>Session ID</strong>
+            </TableCell>
+            <TableCell></TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {submissions.map((submission) => (
+            <CollapsibleRow {...submission} />
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+};
+
+const CollapsibleRow: React.FC<Submission> = (submission) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  return (
+    <React.Fragment key={submission.eventId}>
+      <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
+        <TableCell
+          sx={{
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+          }}
+        >
+          <Avatar
+            sx={{
+              background: "none",
+              marginRight: (theme) => theme.spacing(1),
+            }}
+          >
+            {submission.eventType === "Pay" ? (
+              <Payment
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            ) : (
+              <Send
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            )}
+          </Avatar>
+          {submission.eventType}
+        </TableCell>
+        <TableCell>
+          {format(new Date(submission.createdAt), "dd/MM/yy hh:mm:ss")}
+        </TableCell>
+        <TableCell>{submission.sessionId}</TableCell>
+        <TableCell>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={() => setOpen(!open)}
+          >
+            {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
```

Comment:
I haven't actually spotted the issue yet in code, but the behaviour here is not quite right.

When I open the accordion for a Uniform submission, this button disappears and can't be closed.


https://github.com/theopensystemslab/planx-new/assets/20502206/0a7af909-07a1-40ea-baee-6582ccd5dd90

Reply:
The property needed here is `white-space: pre-wrap` which took a bit of digging. I've added to an incoming PR for style update